### ISSUE TITLE
Build fails on GCC 10

### ISSUE THREAD

**Issue by Gaelan** [kcat/openal-soft#519]
GCC 10 contains a [breaking change](https://gcc.gnu.org/gcc-10/porting_to.html): it defaults to `-fno-common`, which requires globals to be declared in exactly one C file, and referenced elsewhere with `extern`. This causes `openal-soft` to fail to build with a linker error:

```
ld: CMakeFiles/OpenAL.dir/Alc/ALu.c.o:(.rodata+0x3bf60): multiple definition of `FuMa2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf40): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/ALu.c.o:(.rodata+0x3bfa0): multiple definition of `SN3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf80): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/ALu.c.o:(.rodata+0x3bfe0): multiple definition of `N3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xfc0): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/bformatdec.c.o:(.rodata+0x2e0): multiple definition of `SN3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf80): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/bformatdec.c.o:(.rodata+0x320): multiple definition of `N3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xfc0): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/bformatdec.c.o:(.rodata+0x2a0): multiple definition of `FuMa2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf40): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/panning.c.o:(.rodata+0x1360): multiple definition of `FuMa2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf40): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/panning.c.o:(.rodata+0x13e0): multiple definition of `N3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xfc0): first defined here
ld: CMakeFiles/OpenAL.dir/Alc/panning.c.o:(.rodata+0x13a0): multiple definition of `SN3D2N3DScale'; CMakeFiles/OpenAL.dir/Alc/ALc.c.o:(.rodata+0xf80): first defined here
collect2: error: ld returned 1 exit status
```

This could be fixed by declaring globals in exactly one place as noted above, or by explicitly passing `-fcommon` to gcc.

**Comment by kcat** [comment]
How is it supposed to work for global `constexpr` values/arrays? Inline variables are a C++17 feature, but it should be possible to have global `constexpr` variables in C++14.

**Comment by Gaelan** [comment]
I'm afraid I know very little about C++, so I can't be of much help here.

**Comment by kcat** [comment]
Fixed with commit 20f5e7c1fafcd40cdee2f7226ae81480ae049d1a.
